src/transaction/log_applier_sql_log.c

```diff
@@ -171,15 +173,45 @@ sl_init (const char *db_name, const char *repl_log_path)
 {
   char tmp_log_path[PATH_MAX];
   char basename_buf[PATH_MAX];
+  char er_msg[PATH_MAX];
+  const char *sql_log_path = NULL;
 
   memset (&sl_Info, 0, sizeof (sl_Info));
 
-  snprintf (tmp_log_path, PATH_MAX, "%s/sql_log/", repl_log_path);
-  create_dir (tmp_log_path);
-
   strcpy (basename_buf, repl_log_path);
-  snprintf (sql_log_base_path, PATH_MAX, "%s/sql_log/%s.sql.log", repl_log_path, basename (basename_buf));
-  snprintf (sql_catalog_path, PATH_MAX, "%s/%s_applylogdb.sql.info", repl_log_path, db_name);
+
+  sql_log_path = prm_get_string_value (PRM_ID_HA_SQL_LOG_PATH);
+  if (sql_log_path == NULL)
+    {
+      sql_log_path = repl_log_path;
+    }
+
+  if (!IS_ABS_PATH (sql_log_path))
```

Comment:
How about using `realpath()` ?

Reply:
I explained why I don't use the realpath(). (The limitation of realpath() is that the path set in the ha_sql_log_path should be existed.) I consider one more time there is a good way to handle this issue.